test/Microsoft.Framework.ApplicationHost.FunctionalTests/AppHostTests.cs

```diff
@@ -170,5 +171,131 @@ public void AppHostShowsErrorWhenCurrentTargetFrameworkWasNotFoundInProjectJson(
+                {
+                    foreach (var command in commands)
+                    {
+                        data.Add(component[0], component[1], component[2], command.Key, command.Value);
+                    }
+                }
+
+                return data;
+            }
+        }
+
+        [ConditionalTheory]
+        [MemberData(nameof(ClrRuntimeComponentsAndCommandsSet))]
+        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        public void AppHost_ExecutesCommands(
+            string flavor,
+            string os,
+            string architecture,
+            string command,
+            string expectedOutput)
+        {
+            var projectName = "Project Name";
+            var projectStructure =
+$@"{{
+  '.': ['Program.cs', '{ Project.ProjectFileName }']
+}}";
+            var programContents =
+@"using System;
+
+namespace Project_Name
+{
+    public class Program
+    {
+        public void Main(string[] arguments)
+        {
+            for (var i = 0; i < arguments.Length; i++)
+            {
+                var argument = arguments[i];
+                if (!string.IsNullOrWhiteSpace(argument))
+                {
+                    Console.WriteLine($""{ i }: '{ argument }'"");
+                }
+            }
+        }
+    }
+}";
+            var projectJsonContents =
+$@"{{
+  ""commands"": {{
+    ""one"": ""\""{ projectName }\"" one two"",
+    ""two"": ""\""{ projectName }\"" ^>three &&>>^\"""",
+    ""three"": ""\""{ projectName }\"" four \""argument five\""""
+  }},
+  ""frameworks"" : {{
+    ""dnx451"": {{ }}
+  }}
+}}";
+
+            using (var applicationRoot = TestUtils.CreateTempDir())
+            {
+                var projectPath = Path.Combine(applicationRoot, projectName);
+                DirTree.CreateFromJson(projectStructure)
+                    .WithFileContents("Program.cs", programContents)
+                    .WithFileContents(Project.ProjectFileName, projectJsonContents)
+                    .WriteTo(projectPath);
+                var runtimeHomePath = _fixture.GetRuntimeHomeDir(flavor, os, architecture);
+
+                var exitCode = DnuTestUtils.ExecDnu(
+                    runtimeHomePath,
+                    subcommand: "restore",
+                    arguments: null,
+                    environment: null,
+                    workingDir: projectPath);
+                Assert.Equal(0, exitCode); // Guard
+
+                string output;
+                string error;
+                exitCode = BootstrapperTestUtils.ExecBootstrapper(
+                    runtimeHomePath,
+                    arguments: $@"""{ projectPath }"" { command } extra",
+                    stdOut: out output,
+                    stdErr: out error);
+
+                Assert.Equal(0, exitCode);
+                Assert.Empty(error);
+                Assert.Equal(expectedOutput, output);
```

Comment:
`DNX_TRACE` is turned on by default in our TeamCity CI and you'll get extra info in output.

Use `Assert.Contains()` or turn off `DNX_TRACE` in the subprocess with something like:
https://github.com/aspnet/dnx/blob/dev/test/Bootstrapper.FunctionalTests/BootstrapperTests.cs#L114